src/cplus_plugin/models/constant_raster.py

```diff
@@ -0,0 +1,432 @@
+            return "_".join(parts)
+        return self.alias_name or self.component_id
+
+    def to_dict(self) -> dict:
+        """Serialize to dictionary."""
+        return {
+            "value_info": self.value_info.to_dict() if self.value_info else {},
+            "component_uuid": str(self.component.uuid) if self.component else "",
+            "prefix": self.prefix,
+            "base_name": self.base_name,
+            "suffix": self.suffix,
+            "alias_name": self.alias_name,
+            PATH_ATTRIBUTE: self.path,
+            "skip_raster": self.skip_raster,
+            "enabled": self.enabled,
+            "component_id": self.component_id,
+            "component_type": (
+                self.component_type.value
+                if self.component_type
+                else ModelComponentType.UNKNOWN.value
+            ),
+        }
+
+    @staticmethod
+    def from_dict(
+        d: dict, component_lookup: typing.Callable[[str], LayerModelComponent]
+    ) -> "ConstantRasterComponent":
+        """Deserialize from dictionary.
+
+        :param d: Dictionary data
+        :param component_lookup: Function to retrieve component by UUID
+        :returns: ConstantRasterComponent instance
+        """
+        component = None
+        component_uuid = d.get("component_uuid")
+        if component_uuid:
+            component = component_lookup(component_uuid)
+
+        value_info_data = d.get("value_info", {})
+        value_info = (
+            ConstantRasterInfo.from_dict(value_info_data)
+            if value_info_data
+            else ConstantRasterInfo()
+        )
+
+        component_type_str = d.get("component_type", ModelComponentType.UNKNOWN.value)
+        component_type = ModelComponentType.from_string(component_type_str)
+
+        return ConstantRasterComponent(
+            value_info=value_info,
+            component=component,
+            prefix=d.get("prefix", ""),
+            base_name=d.get("base_name", ""),
+            suffix=d.get("suffix", ""),
+            alias_name=d.get("alias_name", ""),
+            path=d.get(PATH_ATTRIBUTE, ""),
+            skip_raster=bool(d.get("skip_raster", True)),
+            enabled=bool(d.get("enabled", True)),
+            component_id=d.get("component_id", ""),
+            component_type=component_type,
+        )
+
+    def to_map_layer(self) -> typing.Optional[QgsRasterLayer]:
+        """Convert to QGIS raster layer.
+
+        :returns: QgsRasterLayer if path exists and is valid, None otherwise
+        """
+        if not self.path or not self.path.strip():
+            return None
+
+        if self.qgis_map_layer:
+            return self.qgis_map_layer
+
+        if os.path.exists(self.path):
+            layer = QgsRasterLayer(self.path, self.alias_name or self.component_id)
+            if layer.isValid():
+                self.qgis_map_layer = layer
+                return layer
+
+        return None
+
+    def to_pwl(self):
```

Comment:
@osundwajeff remove this function as I realized its not structurally correct in the design since PWLs are not applicable for activity-based components.